feat(bi): add business intelligence reporting tool with CRM integration

emmdim opened the pull request:
IMPROTANT: Will probably move to a separate repo to make it a `function` in the hosting service

Add a new BI command-line tool that generates comprehensive organization reports by aggregating data from MongoDB database and Vochain API. The tool provides:

- Organization analytics comparing database and blockchain data
- Lead generation and export functionality
- CRM integration with Holded API for contact and lead management
- Automated email reporting via SMTP
- JSON export of business intelligence reports
- Filtering of internal vocdoni.org domain organizations

This tool enables automated business intelligence workflows for tracking organization onboarding, activity, and lead conversion metrics.

 Closes #327

altergui commented:
this doesn't look ready for review, right?
re-request review when it's ready, or mark as Draft and re-request now, then mark as Ready when it's ready

emmdim committed:
feat(bi): add business intelligence reporting tool with CRM integration

Add a new BI command-line tool that generates comprehensive organization
reports by aggregating data from MongoDB database and Vochain API. The
tool provides:

- Organization analytics comparing database and blockchain data
- Lead generation and export functionality
- CRM integration with Holded API for contact and lead management
- Automated email reporting via SMTP
- JSON export of business intelligence reports
- Filtering of internal vocdoni.org domain organizations

This tool enables automated business intelligence workflows for tracking
organization onboarding, activity, and lead conversion metrics.

altergui reviewed (approved):
LGTM, flagged one bug, the rest are comments

altergui reviewed (commented):
.

altergui reviewed (commented):
Inline review comments

emmdim committed:
Solves PR comments
